apps/gateway-ui/src/App.tsx

```diff
@@ -21,7 +21,7 @@ export const UNIT_OPTIONS = ['msats', 'sats', 'btc'] as const;
 export type Unit = (typeof UNIT_OPTIONS)[number];
 
 export const App = React.memo(function Admin(): JSX.Element {
-  const gateway = useMemo(() => new GatewayApi(), []);
+  const gateway = useRef(() => new GatewayApi());
```

Comment:
Note for reviewer: `useRef` better indicates that `gateway` doesn't change, so we can safely remove it as a dep from the `useEffect`s below